FEM-2259 Protect NPE when ad is not coming and going to bg/fg

giladna opened the pull request.

giladna committed:
no message

noamtamim commented on imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/ExoPlayerWithAdPlayback.java:
```diff
@@ -480,6 +481,10 @@ public void setAdCuePoints(AdCuePoints adCuePoints) {
 
     private void initializePlayer(String adUrl, boolean adShouldAutoPlay) {
+        if (TextUtils.isEmpty(adUrl)) {
+            throw new IllegalArgumentException("Ad playback url cannot be empty or null");
```
This will crash the app. I doubt it's a good idea.

giladna replied:
Done thanks!

giladna committed:
send Ad Error instead of Exception + NPE protection

=>Ad Error: VIDEO_PLAY_ERROR with message Error, Ad playback url cannot be empty or null

giladna committed:
extract to a method
